### PR TITLE
fix(link-list): fixed wrong column count and styling on knob change

### PR DESCRIPTION
**Pull request by RaphaelAmadeu**
### Related Ticket(s)

#5040 

### Description

Solved a bug when `LinkList` at `end` variant would split in two columns with 3 elements, when it only should split after 4 items.

Corrected styling on `dds-link-list-item-cta` on `end` variant, so it has the same look as `dds-link-list-item`.

Corrected a Storybook bug that made the items on `three-columns` variant break if `cta-type` changed on knobs.

### CODE REVIEW

File: packages/web-components/src/components/link-list/link-list.ts [hunk 1]
```diff
@@ -119,6 +122,7 @@ class DDSLinkList extends StableSelectorMixin(LitElement) {
   }
 
   updated() {
+    console.log(this.type);
```
Comment (asudoh) [inline packages/web-components/src/components/link-list/link-list.ts:125]:
```suggestion
    console.log(this.type);
```

**Review by asudoh — COMMENTED** [review]
Just one thing, thanks @RaphaelAmadeu!

**Review by asudoh — APPROVED** [review]
Thanks @RaphaelAmadeu for the update!

**Review by kennylam — APPROVED** [review]
LGTM thanks @RaphaelAmadeu!

**Pull request merged**